File: frontend/src/utils/propertyDisplayName.js
export function getPropertyDisplayName(property) {
  const p = property || {};
  const explicit = String(
    p.nickname || p.name || p.property_name || p.property_label || ''
  ).trim();
  if (explicit) return explicit;

  const line1 = String(p.address_line_1 || '').trim();
  const city = String(p.city || p.town || '').trim();
  const postcode = String(p.postcode || '').trim();

  if (line1 && city) return `${line1}, ${city}`;
  if (line1 && postcode) return `${line1}, ${postcode}`;
  if (line1) return line1;
  if (postcode) return postcode;
  return 'Unnamed property';
}

File: backend/presentation/property_display_name.py
"""Shared property display-name resolver for client-facing text."""

from __future__ import annotations

from typing import Any, Dict


def get_property_display_name(property_row: Dict[str, Any] | None) -> str:
    row = property_row or {}
    explicit = str(
        row.get("nickname")
        or row.get("name")
        or row.get("property_name")
        or row.get("property_label")
        or ""
    ).strip()
    if explicit:
        return explicit

    line1 = str(row.get("address_line_1") or "").strip()
    city = str(row.get("city") or row.get("town") or "").strip()
    postcode = str(row.get("postcode") or "").strip()

    if line1 and city:
        return f"{line1}, {city}"
    if line1 and postcode:
        return f"{line1}, {postcode}"
    if line1:
        return line1
    if postcode:
        return postcode
    return "Unnamed property"

